lib/UI/calendar/widgets/logout_button.dart

```diff
@@ -11,34 +12,19 @@ class LogoutButton extends StatefulWidget {
   State<LogoutButton> createState() => _LogoutButtonState();
 }
 
-class _LogoutButtonState extends State<LogoutButton> {
+class _LogoutButtonState extends State<LogoutButton> with Dialogs {
   bool _isLoading = false;
 
   @override
   Widget build(BuildContext context) {
     return Padding(
       padding: const EdgeInsets.only(right: 16.0),
       child: InkWell(
-        onTap: () async {
-          if (!_isLoading) {
-            setState(() => _isLoading = true);
-            await FirebaseAuth.instance.signOut();
-            setState(() => _isLoading = false);
-            _navigateToLandingScreen(context);
-          }
-        },
+        onTap: _isLoading ? showSignoutConfirmationDialog : null,
         child: _isLoading
             ? Center(child: CircularProgressIndicator(strokeWidth: 2))
             : Icon(Icons.logout),
```

Comment:
The check here is also irrelevant since nothing is toggling _isLoading. Removed that as well.